### PR TITLE
Fix more warnings in calculator example.

### PR DESCRIPTION
**Pull request by trombonehero**
These commits contain slightly more debatable fixes to warnings like `exit-time-destructor` and `weak-vtables`.

### CODE REVIEW

File: examples/calculator/calculator.cc [hunk 1]
```diff
@@ -7,11 +7,8 @@ using namespace std;
 using namespace pegmatite;
 
 
-/**
- * The AST namespace contains the abstract syntax tree for this grammar.  
- */
-namespace AST
```
Comment (davidchisnall) [inline examples/calculator/calculator.cc]: Please can we keep the AST and Parser namespaces?  They won't generate the most efficient code, but they help document the structure of the code (which, for an example, is probably more important).

File: examples/calculator/calculator.cc [hunk 2]
```diff
@@ -158,7 +152,7 @@ struct CalculatorGrammar
 	 */
 	static const CalculatorGrammar& get()
 	{
-		static CalculatorGrammar g;
+		static CalculatorGrammar& g = *new CalculatorGrammar();
```
Comment (davidchisnall) [inline examples/calculator/calculator.cc]: I don't really like this idiom.  I can see it's useful where we care about performance more than we care about memory leaks, but it's probably bad for an example to contain code that teaches people to leak memory.

**Comment by trombonehero** [comment]
These are all fair enough points... perhaps we should just disable these warnings in the examples directory.

**Comment by trombonehero** [comment]
Are `CMAKE_CXX_FLAGS` set within a subdirectory global, or just within that subdirectory? I think it's fair to say that we won't care about exit-time destructors or weak vtables in any future examples.

**Comment by davidchisnall** [comment]
I believe that `CMAKE_CXX_FLAGS` is global.  CMake doesn't have sensible namespaces other than the explicit project and file ones.  We should probably set something like `EXAMPLES_CXX_FLAGS` in the examples directory and then set [`target_compile_options`](https://cmake.org/cmake/help/v3.6/command/target_compile_options.html#command:target_compile_options) from it.

**Comment by davidchisnall** [comment]
(Also, please squash these commits before pushing, or we will end up with some very confusing history)

**Comment by trombonehero** [comment]
It looks like one needs to set `PARENT_SCOPE` in the `set` command in order to make the changes global:
https://cmake.org/cmake/help/v3.0/command/set.html

**Comment by davidchisnall** [comment]
Ah, that's fine then.

**Pull request merged**